src/heaps/minmax_heap.jl

```diff
@@ -0,0 +1,261 @@
+################################################
+#
+# minmax heap type and constructors
+#
+################################################
+
+mutable struct BinaryMinMaxHeap{T} <: AbstractMinMaxHeap{T}
+    valtree::Vector{T}
+    
+    function BinaryMinMaxHeap{T}() where {T}
+        new{T}(Vector{T}())
+    end
+    
+    function BinaryMinMaxHeap{T}(xs) where {T}
+        valtree = _make_binary_minmax_heap(T, xs)
+        new{T}(valtree)
+    end
+end
+
+binary_minmax_heap(vt::Type{T}) where {T} = BinaryMinMaxHeap{T}()
+binary_minmax_heap(xs::AbstractVector{T}) where {T} = BinaryMinMaxHeap{T}(xs) 
```

Comment:
This is to match the API for constructing the other heap types: `binary_{min|max}heap`. I agree they are strictly speaking not necessary, since we aren't implicitly passing an ordering (or `comparer` as the case may be) like the other heap methods do, but it would frustrate me as a user to have binary heaps be instantiated with `binary_minheap` and minmax heaps with `BinaryMinMaxHeap`.

Reply:
That is a flaw in them because they are really exceptionally old code.
They were made originally in julia 0.1, when things were different.
https://github.com/JuliaCollections/DataStructures.jl/issues/469

Reply:
I've no issue renaming and reorganizing to align with current Julia features and conventions. My main interest was just making sure the heap interfaces were consistent. 

Maybe we can decide what that interface should be for `MinMaxHeap` and then extend it to the rest of the heaps in another PR?

Reply:
Yes, that is what I suggest.
Do this right and we will resolve #469 seperately.

Reply:
I removed the pseudoconstructors and left one inner and one outer constructor:
- `BinaryMinMaxHeap(xs::AbstractVector{T})`
- `BinaryMinMaxHeap(vt::Type{T})`